api/storage_service.cc

```diff
@@ -127,6 +127,21 @@ bool validate_bool(const sstring& param) {
     }
 }
 
+bool validate_bool_x(const sstring& param, bool default_value) {
```

Comment:
Since it has a different signature, it can be simply called `validate_bool` as well.
And if the default_value would be an optional, default initialized to std::nullopt, and the return value be an optional too, maybe both functions can be unified

Reply:
Agree. However, I found 8 different ways of parsing a string into a bool in our API handlers. I planned to generalize it to some extent as a followup.